wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController/ConversationInputBarViewController.swift

```diff
@@ -615,10 +615,43 @@ final class ConversationInputBarViewController: UIViewController,
 
     // MARK: - PingButton
 
-    @objc
-    private func pingButtonPressed(_ button: UIButton?) {
-        appendKnock()
-    }
+        private func confirmPing(completion: @escaping (_ completion: Bool) -> Void) {
+            let participantCount = conversation.localParticipantsCount - 1
+            let title = L10n.Localizable.Conversation.Ping.ManyParticipantsConfirmation.title(participantCount)
+
+            let controller = UIAlertController(
+                title: title,
+                message: nil,
+                preferredStyle: .alert
+            )
+
+            controller.addAction(.cancel { completion(false) })
+
+            let sendAction = UIAlertAction(
+                title: L10n.Localizable.Conversation.Ping.Action.title,
+                style: .default,
+                handler: { _ in completion(true) }
+            )
+
+            controller.addAction(sendAction)
+            self.present(controller, animated: true)
+        }
+
+        @objc
+        private func pingButtonPressed(_ button: UIButton?) {
+            let participantCount = conversation.localParticipantsCount - 1
+            let confirmPingParticipantsLimit = 4
```

Comment:
> Opinion: We could move this constant to top of the file or some constant file.

I'm planning to do that on a separate PR, since I'm going to make more changes on the ConversationInputBarViewController as a whole. @theswiftway 

> Suggestion: confirmPingParticipantsLimit is this like minimumPingParticipants ?

Yes, it is. I really like the suggestion I will make the change.